### PR TITLE
Adding skipped tests for broken alias queries 

### PR DESCRIPTION
**Pull request by fulghum**
Capturing some of the edge cases I found while testing aliases as skipped enginetests. 

I've found where a few of these are coming from, but still need to figure out the best way to fix them. @max-hoffman – if you have any intuition/thoughts/hints on the specific failing tests, I'd love to hear it.

### CODE REVIEW

File: enginetest/queries/column_alias_queries.go [hunk 1]
```diff
@@ -20,6 +20,56 @@ import (
...
+	{
+		// Fails with an unresolved *plan.Project node error
+		// The second Project in the union subquery doens't seem to get its alias reference resolved
+		Query:    `SELECT 1 as a, (select a union select a) as b;`,
+		Expected: []sql.Row{{1, 1}},
+	},
+	{
+		// GMS executes this query, but it is not valid because of the forward ref of alias b.
+		// GMS should return an error about an invalid forward-ref.
+		Query:    `select 1 as a, (select b), 0 as b;`,
```
Comment (max-hoffman) [inline enginetest/queries/column_alias_queries.go:55]: forward reference errors are new to me, but seem like even more of a reason to resolve symbols in a more granular way

File: enginetest/queries/column_alias_queries.go [hunk 2]
```diff
@@ -20,6 +20,56 @@ import (
...
+		// GMS returns the error "found HAVING clause with no GROUP BY", but MySQL executes
+		// this query without any problems.
+		Query:    "select t1.i as a from mytable as t1 having a = t1.i;",
+		Expected: []sql.Row{{1}, {2}, {3}},
+	},
+	{
+		// GMS returns "expression 'dt.two' doesn't appear in the group by expressions", but MySQL will execute
+		// this query. It does seem odd that we are not selecting dt.two though. Perhaps MySQL sees those values are
+		// always going to be literals and optimizes?
+		Query:    "select 1 as a, one + 1 as mod1, dt.* from mytable as t1, (select 1, 2 from mytable) as dt (one, two) where dt.one > 0 group by one;",
```
Comment (max-hoffman) [inline enginetest/queries/column_alias_queries.go:68]: The way we separate sorts, having, groupby, etc makes it hard to resolve group by's select and grouping expressions. I collapsed limit, order by, and distinct into union for the same reason I think collapsing our general tree structure into flatter scopes would make it easier to resolve symbols.

File: enginetest/queries/column_alias_queries.go [hunk 3]
```diff
@@ -20,6 +20,56 @@ import (
...
+		// The dual table's schema can conflict with aliases
+		//
+		Query:    `select "foo" as dummy, (select dummy)`,
+		Expected: []sql.Row{{"foo", "foo"}},
+	},
+	{
+		// This query crashes Dolt with "panic: unexpected type x"
+		// Seems likely to be coming from dual table's schema and dummy value.
+		// https://github.com/dolthub/dolt/issues/4256
+		Query:    `SELECT *, (select i union select i) as a from mytable;`,
```
Comment (max-hoffman) [inline enginetest/queries/column_alias_queries.go:34]: also new to be but concerning, can't believe we panic on this

File: enginetest/queries/column_alias_queries.go [hunk 4]
```diff
@@ -20,6 +20,56 @@ import (
...
+		// aliases by name and the second a alias overwrites the one that should be used. GMS should not allow
+		// the subquery to see the second "as a" alias, since that expression should only see aliases defined
+		// in previous expressions (i.e. forward-references are disallowed).
+		Query:    `SELECT 1 as a, (select a) as a;`,
+		Expected: []sql.Row{{1, 1}},
+	},
+	{
+		// Fails with an unresolved *plan.Project node error
+		// The second Project in the union subquery doens't seem to get its alias reference resolved
+		Query:    `SELECT 1 as a, (select a union select a) as b;`,
```
Comment (max-hoffman) [inline enginetest/queries/column_alias_queries.go:49]: i would have thought this one passed, unless we aren't passing through the parent scope

**Comment by fulghum** [comment]
Closing this one now that these problematic queries are tracked in PR https://github.com/dolthub/go-mysql-server/pull/1281

**Pull request closed without merging**